L1Trigger/L1CaloTrigger/plugins/Phase2L1CaloEGammaEmulator.cc

```diff
@@ -597,20 +605,35 @@ void Phase2L1CaloEGammaEmulator::produce(edm::Event& iEvent, const edm::EventSet
   auto L1GCTTowers = std::make_unique<l1tp2::CaloTowerCollection>();
   auto L1GCTFullTowers = std::make_unique<l1tp2::CaloTowerCollection>();
   auto L1GCTEGammas = std::make_unique<l1t::EGammaBxCollection>();
+  auto L1DigitizedClusterCorrelator = std::make_unique<l1tp2::DigitizedClusterCorrelatorCollection>();
+  auto L1DigitizedTowerCorrelator = std::make_unique<l1tp2::DigitizedTowerCorrelatorCollection>();
+  auto L1DigitizedClusterGT = std::make_unique<l1tp2::DigitizedClusterGTCollection>();
 
   //----------------------------------------------------
   // Apply the GCT firmware code to each GCT card
   //----------------------------------------------------
 
   for (unsigned int gcc = 0; gcc < p2eg::N_GCTCARDS; gcc++) {
-    p2eg::algo_top(
-        gctCards[gcc], gctToCorr[gcc], gcc, L1GCTClusters, L1GCTTowers, L1GCTFullTowers, L1GCTEGammas, calib_);
+    p2eg::algo_top(gctCards[gcc],
+                   gctToCorr[gcc],
+                   gcc,
+                   L1GCTClusters,
+                   L1GCTTowers,
+                   L1GCTFullTowers,
+                   L1GCTEGammas,
+                   L1DigitizedClusterCorrelator,
+                   L1DigitizedTowerCorrelator,
+                   L1DigitizedClusterGT,
+                   calib_);
   }
 
   iEvent.put(std::move(L1GCTClusters), "GCT");
   iEvent.put(std::move(L1GCTTowers), "GCT");
   iEvent.put(std::move(L1GCTFullTowers), "GCTFullTowers");
   iEvent.put(std::move(L1GCTEGammas), "GCTEGammas");
+  iEvent.put(std::move(L1DigitizedClusterCorrelator), "GCTDigitizedClusterToCorrelator");
```

Comment:
@skkwan do we really need separate collections for the digitized objects? Wouldn't it be easier for the downstream consumers to add the digitized object as data member of the CMSSW (float) counter-part? This is what is done for example in:
https://github.com/cms-l1t-offline/cmssw/pull/1088/files#diff-b728d605f568ab3466237040728947ab6e564a9e94d237d40803fc1cd8fe411dR55

and 

https://github.com/cms-sw/cmssw/blob/master/DataFormats/L1TCorrelator/interface/TkEm.h#L66

Reply:
This is fine for temporary usage as long as it produces the proper content. We can clean this up and reduce the output of the emulator when we are happy that it is functioning as intended.